src/main/java/com/xiaomi/infra/pegasus/rpc/async/ReplicaSession.java

```diff
@@ -119,6 +121,17 @@ public int asyncSend(
     return entry.sequenceId;
   }
 
+  public void sendNegoMsg(negotiation_request msg, long timeoutInMilliseconds) {
+    final RequestEntry entry = new ReplicaSession.RequestEntry();
+    entry.sequenceId = seqId.getAndIncrement();
+    entry.op = new negotiation_operator(msg);
+    entry.callback = new Negotiation.RecvHandler((negotiation_operator) entry.op, this);
+    entry.timeoutTask = addTimer(entry.sequenceId, timeoutInMilliseconds);
+    pendingResponse.put(entry.sequenceId, entry);
+
+    write(entry, fields);
```

Comment:
Why re-use `asyncSend`?

Reply:
asyncSend need the connect state to be `CONNECTED`, but in the duration of negotiation, the connect state is not `CONNECTED`

Reply:
And `sendNegoMsg` should be `synchronized` and double-check connect-state to avoid re-send?